dialer.go

```diff
@@ -590,14 +598,19 @@ type instrumentedConn struct {
 	closed    bool
 	dialerID  string
 	connName  string
+
+	bytesRead    int64
+	bytesWritten int64
+	reportTicker *time.Ticker
+	stopReporter func()
 }
 
 // Read delegates to the underlying net.Conn interface and records number of
 // bytes read
 func (i *instrumentedConn) Read(b []byte) (int, error) {
 	bytesRead, err := i.Conn.Read(b)
 	if err == nil {
-		go trace.RecordBytesReceived(context.Background(), int64(bytesRead), i.connName, i.dialerID)
+		atomic.AddInt64(&i.bytesRead, int64(bytesRead))
```

Comment:
Shall we use https://pkg.go.dev/sync/atomic#Int64.Add instead here?

Reply:
Ditto below.

Reply:
There's apparently a bug around AddInt64 for 32-bit platforms, but we can address this in a separate PR.